### PR TITLE
refactor: extract timerange calculations

### PR DESCRIPTION
**Pull request by alamb**
This PR proposes extracting the timerange calculations from the compactor algorithm as it is used in several places (such as 
https://github.com/influxdata/influxdb_iox/pull/6852#discussion_r1096526179)

I think this not only avoids copy/paste it also encapsulates the functionality and ensures the core of the algorithms calculation is clear

### CODE REVIEW

File: compactor2/src/components/files_split/target_level_non_overlap_split.rs [hunk 1]
```diff
@@ -84,20 +80,13 @@ impl FilesSplit for TargetLevelNonOverlapSplit {
             }
         }
 
-        if prev_level_files.is_empty() {
+        // compute time range of prev_level_files
```
Comment (alamb) [inline compactor2/src/components/files_split/target_level_non_overlap_split.rs:83]: This just moves the mechanics of computing the min/max time ranges for a group of parquet files into a structure

File: compactor2/src/components/files_split/target_level_non_overlap_split.rs [hunk 2]
```diff
@@ -107,11 +96,13 @@ impl FilesSplit for TargetLevelNonOverlapSplit {
 
         // Closure that checks if a file overlaps with any prev_level_files and add it
         // to overlapping_files or non_overlapping_files accordingly. Return true if ovelapping
+        let mut non_overlapping_files = Vec::with_capacity(num_files);
```
Comment (alamb) [inline compactor2/src/components/files_split/target_level_non_overlap_split.rs:99]: Moves the declaration of these vectors closer to the code that creates them

File: compactor2/src/file_group.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
+//! Utilities for working with groups of [`ParquetFile`]
```
Comment (alamb) [inline compactor2/src/file_group.rs:1]: I think there are some other common patterns I want to put in here over time

File: compactor2/src/file_group.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    min_time: Timestamp,
+    /// max of max_time, across all files
+    max_time: Timestamp,
+}
+
+impl FilesTimeRange {
+    /// Computes the min/max times across all `files`. Returns `None` if there are no files
+    pub fn try_new<'a>(files: impl IntoIterator<Item = &'a ParquetFile>) -> Option<Self> {
+        let mut files = files.into_iter();
+        let first_file = files.next()?;
```
Comment (NGA-TRAN) [inline compactor2/src/file_group.rs:19]: TIL

File: compactor2/src/file_group.rs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        // fully within
+        assert!(range.contains(&ParquetFileBuilder::new(4).with_time_range(100, 250).build()));
+
+        // on edge going out
+        assert!(range.contains(&ParquetFileBuilder::new(4).with_time_range(250, 300).build()));
+        // on edge
+        assert!(range.contains(&ParquetFileBuilder::new(4).with_time_range(300, 350).build()));
+        // too high
+        assert!(!range.contains(&ParquetFileBuilder::new(4).with_time_range(301, 350).build()));
+    }
```
Comment (NGA-TRAN) [inline compactor2/src/file_group.rs:75]: Nice tests

File: compactor2/src/components/files_split/target_level_non_overlap_split.rs [hunk 6]
```diff
@@ -107,11 +96,13 @@ impl FilesSplit for TargetLevelNonOverlapSplit {
...
+        let mut overlapping_files = Vec::with_capacity(num_files);
         let mut check_overlapping_and_add = |file: ParquetFile| -> bool {
             let mut overlapping = false;
 
             // Check if file overlaps with (min_time, max_time)
-            if file.min_time <= max_time && file.max_time >= min_time {
+            if prev_level_range.contains(&file) {
```
Comment (NGA-TRAN) [inline compactor2/src/components/files_split/target_level_non_overlap_split.rs:105]: 👍

**Review by NGA-TRAN — APPROVED** [review]
Nice. Thanks for refactoring this @alamb

**Pull request merged**